packages/utilities/src/properties.ts

```diff
@@ -271,7 +271,9 @@ export function getNativeProps<T>(props: {}, allowedPropNames: string[], exclude
     (propName: string) => {
       return (
         (!excludedPropNames || excludedPropNames.indexOf(propName) < 0) &&
-        (propName.indexOf('data-') === 0 || propName.indexOf('aria-') === 0 || allowedPropNames.indexOf(propName) >= 0)
+        (propName.indexOf('data-') === 0 || propName.indexOf('aria-') === 0 || allowedPropNames.indexOf(propName) >= 0) &&
+        // tslint:disable-next-line:no-any
+        (props as any)[propName] !== undefined
```

Comment:
This is very subtle and I'm worried about unknown side effects. The main question I have is: would we expect anyone to be purposely setting `undefined` values to native props to override defined values?

Reply:
I need to dig in to understand why need to make a change here but you should definitely be putting the motivation in code comments here.

Reply:
I explained the need in the PR description but I'll add code comments as well.

Reply:
I saw it, I just didn't grasp it :) I need to dig deeper to try to grasp it.

Reply:
In `DatePicker`, I added a`id={labelId}` prop to use of `TextField`, which can be `undefined` since the prop is optional. This caused:
1. [This call](https://github.com/OfficeDev/office-ui-fabric-react/blob/56a233603a718e7a361d239a28a56ed91a1ddd84/packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx#L411) to `getNativeProps` to have an explicit property in its return object of `id: undefined`.
1. [This spread](https://github.com/OfficeDev/office-ui-fabric-react/blob/56a233603a718e7a361d239a28a56ed91a1ddd84/packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx#L419) actually causes the `id: undefined` value to overwrite the `id` prop on the line above.
1. Which both caused breakage on the default label as clicking on it no longer causes the `DatePicker` to appear.

I couldn't reorder the prop expansion in `TextField` as we don't want the internal `id` to overwrite any actual defined prop values. To me the best fix was to remove `undefined` property elements from being returned by `getNativeProps` result. This prevents the result from overwriting defined values when spread.